dtls_debug.c

```diff
@@ -107,12 +107,23 @@ dtls_strnlen(const char *s, size_t maxlen) {
   return n;
 }
 
+/**
+ * Write service-address as text to buffer.
+ *
+ * The text is \000 terminated.
+ *
+ * \param addr   The session including the sockaddr.
+ * \param buf    The buffer to write the service address into.
+ * \param len    The actual length of \p buf.
+ * \return Less than zero on error, the number of bytes written otherwise.
+ */
 static size_t
 dsrv_print_addr(const session_t *addr, char *buf, size_t len) {
 #ifdef HAVE_ARPA_INET_H
   const void *addrptr = NULL;
   in_port_t port;
   char *p = buf;
+  int append;
```

Comment:
This should be of type `size_t`.

Reply:
`snprintf`, returns `int`, therefore I chose that.

Reply:
I'm still in doubt, if `size_t` is really better. Or if two locals are required `int` and `size_t`.
Currently I don't see, that the `int` range is exceeded. I also don't see, that `snprintf` returns an error.
My preference is still `int`. but if the other variants, `size_t` or `int` and `size_t` are preferred, I adapt the code.

Reply:
Well, for one, `len` has the type `size_t`, clearly documenting that `len` is for sizes. `append` pretty much deals with sizes as well and relates to `len` in this function. `dtls_strnlen()` returns a `size_t`, hence the result must be treated as `size_t`.
The `snprintf` return type indeed is of type `int`. As we do not know if and why some exotic implementations might to actually return an error here, we should use a second variable of type `int` for this.

Reply:
OK, then go for two locals.